### PR TITLE
Add Wannier90

### PR DESCRIPTION
**Pull request by mfherbst**
Adds Wannier90 as library and executable.

### CODE REVIEW

File: W/wannier90/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/wannier90-*/
+
+shared="-shared"
+if [[ "${target}" == *-apple-* ]]; then
+    shared="-dynamiclib"
+fi
```
Comment (giordano) [inline W/wannier90/build_tarballs.jl:20]: I'm mildly sure that `-shared` works also for macOS libraries
```suggestion
```

File: W/wannier90/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+cat > make.inc << EOF
+    F90    = $FC
+    FCOPTS = $FFLAGS -fPIC -O2
```
Comment (giordano) [inline W/wannier90/build_tarballs.jl:23]: You can use `-O3` if you want, what you can't use is `-march`
```suggestion
    FCOPTS = $FFLAGS -fPIC -O3
```

File: W/wannier90/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    LDOPTS = $LDFLAGS -fPIC
+    LIBS   = -lopenblas -L$prefix/lib/
```
Comment (giordano) [inline W/wannier90/build_tarballs.jl:25]:
```suggestion
    LIBS   = -lopenblas -L${libdir}
```

File: W/wannier90/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    DYNLIBRARYEXTENSION = $dlext
+    SHAREDLIBFLAGS      = $shared
+EOF
+
+make -j${nproc} wannier dynlib
+
+# Installation in the Makefile is broken
+install -d $prefix/bin $prefix/lib
```
Comment (giordano) [inline W/wannier90/build_tarballs.jl:33]:
```suggestion
install -d ${bindir} ${libdir}
```

File: W/wannier90/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+for exe in wannier90; do
+    install -m755 ${exe}.x $prefix/bin/${exe}${exeext}
```
Comment (giordano) [inline W/wannier90/build_tarballs.jl:35]:
```suggestion
    install -m755 ${exe}.x ${bindir}/${exe}${exeext}
```

File: W/wannier90/build_tarballs.jl [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+done
+for lib in libwannier.*; do
+    install -m644 $lib $prefix/lib/$lib
```
Comment (giordano) [inline W/wannier90/build_tarballs.jl:38]:
```suggestion
    install -m644 $lib ${libdir}/$lib
```

File: W/wannier90/build_tarballs.jl (hunk 2 above)
Comment (mfherbst) [reply W/wannier90/build_tarballs.jl:23]: I tried and it caused issues on my local built by architecture-dependent optimisations. Will that not occur on the build farm as well?

File: W/wannier90/build_tarballs.jl (hunk 1 above)
Comment (mfherbst) [reply W/wannier90/build_tarballs.jl]: I'll remove that stuff completely then.

File: W/wannier90/build_tarballs.jl [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libwannier", :libwannier),
+    ExecutableProduct("wannier90", :wannier90)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="OpenBLAS32_jll", uuid="656ef2d0-ae68-5445-9ca0-591084a874a2"))
```
Comment (giordano) [inline W/wannier90/build_tarballs.jl:49]:
```
┌ Warning: To ensure that the correct version of libgfortran is found at runtime, add the following entry to the list of dependencies of this builder
│ 
│     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"))
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/K52TD/src/auditor/compiler_abi.jl:4
```
```suggestion
    Dependency(PackageSpec(name="OpenBLAS32_jll", uuid="656ef2d0-ae68-5445-9ca0-591084a874a2")),
    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
```

Comment (mfherbst) [reply W/wannier90/build_tarballs.jl]: Good catch, thanks ... for some reason I completely ignored this :smile:.

**Pull request merged**